ws/websocket.go

```diff
@@ -89,7 +98,7 @@ type WebSocket struct {
 	connection  *websocket.Conn
 	id          string
 	outQueue    chan []byte
-	closeSignal chan bool
+	closeSignal chan error
```

Comment:
add a comment what this really does: signal the socket to close or signal to consumers that the socket has closed?

Reply:
only used for internal signaling between goroutines (i.e. your second option)